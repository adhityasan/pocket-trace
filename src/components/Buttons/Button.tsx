import { Feather } from "@expo/vector-icons";
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  TouchableOpacityProps,
  View,
  ViewStyle
} from "react-native";

import { COLOR_LIGHT } from "@services/constants/color";

interface IButtonProps extends TouchableOpacityProps {
  label: string;
  icon?: keyof typeof Feather.glyphMap;
  buttonStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  iconStyle?: StyleProp<TextStyle>;
}

const Button = ({
  label,
  icon,
  buttonStyle,
  labelStyle,
  iconStyle,
  ...touchableOpacityProps
}: IButtonProps) => {
  return (
    <TouchableOpacity style={styles.wrapper} {...touchableOpacityProps}>
      <View style={[styles.button, buttonStyle]}>
        {icon ? <Feather name={icon} style={[styles.icon, iconStyle]} /> : null}
        <Text style={[styles.label, labelStyle]}>{label}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    marginVertical: 5
  },
  button: {
    padding: 12,
    backgroundColor: COLOR_LIGHT.PRIMARY,
    borderRadius: 6,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center"
  },
  label: {
    fontSize: 18,
    color: COLOR_LIGHT.LIGHT,
    fontWeight: "500",
    textAlign: "center"
  },
  icon: {
    marginRight: 10,
    fontSize: 18,
    color: COLOR_LIGHT.LIGHT
  }
});

export default Button;
