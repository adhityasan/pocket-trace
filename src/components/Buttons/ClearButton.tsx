import { Feather } from "@expo/vector-icons";
import {
  Pressable,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacityProps,
  View,
  ViewStyle
} from "react-native";

import { COLOR_LIGHT } from "@services/constants/color";

interface IButtonProps extends TouchableOpacityProps {
  label: string;
  icon?: keyof typeof Feather.glyphMap;
  buttonStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  iconStyle?: StyleProp<TextStyle>;
}

const ClearButton = ({
  label,
  icon,
  buttonStyle,
  labelStyle,
  iconStyle,
  ...touchableOpacityProps
}: IButtonProps) => {
  return (
    <Pressable
      style={({ pressed }) => [pressed && styles.wrapper_pressed, styles.wrapper]}
      {...touchableOpacityProps}>
      <View style={[styles.button, buttonStyle]}>
        {icon ? <Feather name={icon} style={[styles.icon, iconStyle]} /> : null}
        <Text style={[styles.label, labelStyle]}>{label}</Text>
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    marginVertical: 5
  },
  wrapper_pressed: {
    backgroundColor: COLOR_LIGHT.SECONDARY,
    borderRadius: 6
  },
  button: {
    padding: 12,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center"
  },
  label: {
    fontSize: 18,
    color: COLOR_LIGHT.PRIMARY,
    fontWeight: "500",
    textAlign: "center"
  },
  icon: {
    marginRight: 10,
    fontSize: 18,
    color: COLOR_LIGHT.PRIMARY
  }
});

export default ClearButton;
