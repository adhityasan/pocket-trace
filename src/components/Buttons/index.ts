export { default as Button } from "./Button";
export { default as ClearButton } from "./ClearButton";
