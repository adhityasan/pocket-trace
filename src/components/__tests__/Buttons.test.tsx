import { create } from "react-test-renderer";

import { Button, ClearButton } from "@components/Buttons";

/**
 * @components/Buttons/Button.tsx
 */
describe("components/Buttons/Button", () => {
  it("renders correctly", () => {
    const tree = create(<Button label="Button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("trigger action on press", () => {
    let isPressed = false;
    const buttonRenderer = create(
      <Button
        label="Button"
        onPress={() => {
          isPressed = true;
        }}
      />
    );
    buttonRenderer.root.props?.onPress();
    expect(isPressed).toBeTruthy();
  });
});

/**
 * @components/Buttons/ClearButton.tsx
 */
describe("components/Buttons/ClearButton", () => {
  it("renders correctly", () => {
    const tree = create(<ClearButton label="Clear Button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("trigger action on press", () => {
    let isPressed = false;
    const buttonRenderer = create(
      <Button
        label="Button"
        onPress={() => {
          isPressed = true;
        }}
      />
    );
    buttonRenderer.root.props?.onPress();
    expect(isPressed).toBeTruthy();
  });
});
