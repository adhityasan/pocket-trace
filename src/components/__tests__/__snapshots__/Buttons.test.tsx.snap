// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Buttons/Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginVertical": 5,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#5145a0",
          "borderRadius": 6,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fafafa",
            "fontSize": 18,
            "fontWeight": "500",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`components/Buttons/ClearButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Object {
        "marginVertical": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#5145a0",
            "fontSize": 18,
            "fontWeight": "500",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Clear Button
    </Text>
  </View>
</View>
`;
